addons/product/static/src/product_catalog/kanban_controller.js

```diff
@@ -25,11 +25,17 @@ export class ProductCatalogKanbanController extends KanbanController {
 
     async _defineButtonContent() {
         // Define the button's label depending of the order's state.
+        const fields = this.orderResModel === 'account.move' ? ["state", "move_type"] : ["state"];
         const orderStateInfo = await this.orm.searchRead(
-            this.orderResModel, [["id", "=", this.orderId]], ["state"]
+            this.orderResModel, [["id", "=", this.orderId]], fields
         );
+
         const orderIsQuotation = ["draft", "sent"].includes(orderStateInfo[0].state);
-        if (orderIsQuotation) {
+        if (orderStateInfo[0]?.move_type ==="out_invoice") {
+            this.buttonString = _t("Back to Invoice");
+        } else if (orderStateInfo[0]?.move_type ==="in_invoice") {
+            this.buttonString = _t("Back to Bill");
+        } else if (orderIsQuotation) {
             this.buttonString = _t("Back to Quotation");
         } else {
             this.buttonString = _t("Back to Order");
```

Comment:
It would be cleaner to make the model specific checks be in overrided functions extended in their specific module. We left it this way during the because Sales Orders and Purchase Orders use the same terms and worked for both when the mixin was moved to the product module. You should never refer to models that aren't included in the module (i.e. `account.move`). Until other modules reuse this mixin, I think it's fine to leave this as the default behavior in product and override it only in account

Reply:
Yeah i thought of making an override for the function but since it was used for both purchase and sale, i decided to it this way but you are right that reference a model outside the model itself is not the best idea, i will change that after my holiday :smile: